### PR TITLE
Fix: Stop unrelated PRs failing CI on tests they never touched (one measured scheduling cause, four real bugs)

### PR DESCRIPTION
**Pull request by cheapsteak**
## What's broken

Pull requests fail CI on tests they did not touch. A PR that changed exactly one markdown file failed the `test` job twice on two different pre-existing tests, then passed on the third attempt. Over the five days before this branch, 60 of 250 runs of the Test workflow went red; of the red `test` jobs that were not a real defect in the PR under development, seven were `ServerShutdownLatchTests` (fixed in #815), two `SocketServerSocketOwnershipTests`, two `TerminalLockedAccessTests` (five test reds), three `ControlModeInput*Tests`, and one each of `WorktreeDeletionQueueTests`, `TerminalViewSessionReclamationTests` (on `main` itself), `EventDrivenTestClockSelfTests` (eight reds in one job), `UpdateLauncherTests`, `AutoCloseSetupTests` (twice), and `TerminalTeardownReapTests`. The ubuntu harness job failed three times in 24 hours on `scripts/test-claude-stub.py`, and the tier-3 quiet pass failed four times on `PeerLinkSupervisorTests`.

## Why it happens

One systemic cause accounts for most of the fast-pass reds; the rest are four distinct, real bugs.

**The systemic cause is scheduling latency per suspension hop, and it is measured, not inferred.** Each fast pass runs every test of its targets as concurrently-runnable tasks in one process on the runner's three cores. In a green run of this workflow (PR #816, attempt 3), the per-test reported duration, which Swift Testing measures from the moment the body starts and is therefore time suspended waiting for a thread, was:

| pass | tests | wall s | p50 | p90 | p99 | max | > 45 s | > 90 s |
|---|---|---|---|---|---|---|---|---|
| 1 (`^TBDDaemonTests\.`) | 5169 | 143 | 65.6 | 82.6 | 92.1 | 95.5 | 4511 | 118 |
| 2 (everything else) | 4836 | 57 | 2.3 | 51.4 | 51.9 | 55.3 | 1054 | 0 |

One hop costs tens of seconds in pass 1, so a handshake that needs k hops costs k times that. `TestDeadlines.saturatedPass` (90 s) sits at p99 of a healthy pass 1; the 45 s clock guards sit below its median; a per-test `.timeLimit(.minutes(1))` in pass 2 sits five seconds above a healthy run's maximum. "0 of 8 latch callers after 90 s" is three to four hops per caller, not a priority story: a probe added by this branch measured the test body, `Task {}`, `Task.detached {}` and the gate executor all running at the same priority (medium, 21) on the CI runner, which refutes the rationale written into the tree by #815. The executor injection there still works, for a different reason: it removes the hops from the shared queue.

**The four distinct bugs:**

- `AutoCloseSetupTests` waited for the setup terminal's row to vanish and then asserted the tab order was pruned. Production deletes the row, then the tab, then rewrites the order in three separate awaits, so the wait returned on an intermediate state. It failed in 0.5 s, twice.
- `TerminalTeardownReapTests` reported "forkpty must have produced both pids" with no cause: SwiftTerm leaves `shellPid` at 0 when `fork` fails and surfaces no errno.
- `scripts/claude-stub.py` lost a stop signal sent in the window between the sandbox appearing and `claude` spawning. The handler raised from a context CPython discards exceptions in (an importlib lock callback, captured stack in the commit), so the wrapper went on to spawn `claude` and waited on it for the rest of the test's 30 s deadline. Reproduced 4 in 980 runs in a 2-CPU Linux container before the fix, 0 in 720 after.
- `PeerLinkSupervisorTests.reconnectBackoffGrowsAcrossRepeatedChildExit` compared two spawn-to-spawn virtual-time samples, each inflated by a real-time-paced term (the previous connection's EOF propagation) that the test's driver converted into virtual advances. The difference had variance the assertion did not allow for; observed values were -0.25, 0.0 and 0.5 against a required 1.0.

**The 71 "known issues" reported on every run** are all deliberate `withKnownIssue` self-tests: 69 from `EventDrivenTestClockSelfTests`, one from `BoundedGateWaitTests`, one from `FlakyQuarantineSelfTests`. Pass 2's 11 are `AppStatePublishFrequencyTests` plus the shared self-tests. Nothing there hides a failure; Tests/CLAUDE.md now says so.

## What this PR does

The rule, and the seams that make it easy to follow:

- **One shared hang guard for the fast pass.** `.fastPassBounded` (Tests/TestSupport/ClockTestSupport.swift) is the single dial, derived from the measured latency; `.clockDriven` is now an alias of it. Every hand-written `.timeLimit(.minutes(1|2|3))` in a fast-pass target moves to it; none of those limits was a regression detector. The tier-3 suites that pin their own one-minute limit as a proof are untouched.
- **Every bounded wait in a fast-pass target takes its deadline from `TestDeadlines`.** About thirty hand-rolled waits with 2 to 30 s literals now use the shared constant or the shared `waitFor`. Negative-assertion windows ("nothing fires within X") and the one discriminating threshold (`FileWatcherTests.attemptWindow`) are deliberately unchanged and say so in place.
- **Executor injection where the test owns the callee**, the remedy #815 established: `ControlModeInputRouter` gains an `executor:` parameter (default nil, both production call sites unchanged), the two fast-pass control-mode suites build it on the tests' own executor and run their reply feed on `gateHoldingTask`, and a pin asserts that delivery, the post-suspension resumption and the actor-side health edge all ran on those threads. This is what took the three control-mode tests, which failed at their 90 s waits with partial progress on a p50-109 s runner, off the saturated queue entirely.
- **`TaskPriorityParityTests`** pins the measured fact the wait rules rest on, with a discriminating control that proves the apparatus reports a probe's own priority. Tests/CLAUDE.md's "Thread-blocking gates" section and the latch test's comments now state the hop-count mechanism instead of the refuted priority claim.

The four repairs:

- `AutoCloseSetupTests` waits on the last write of the teardown (row gone and order pruned); every assertion is unchanged.
- `TerminalTeardownReapTests.startChild` captures errno the instant `startProcess` returns and the failure is a thrown, described error carrying errno, the `/dev/ttys*` count, the open-fd count and `kern.tty.ptmx_max`, so the next occurrence names its cause.
- `scripts/claude-stub.py`: the signal handler never raises; it records, and `raise_if_stopped()` on the main path is the only place `Terminated` is born, at every point between one resource existing and the next. A signal in the binary-not-found window is reported as a stop, and two signals during the exit window are both reported. Deterministic tests fire the signal from the exception-discarding context; the file's harness runs 61 tests.
- `PeerLinkSupervisor` gains a `jitter` seam (defaulted to the production ±20%, sole production call site unchanged); the test pins it to zero, asserts the requested delays are exactly `[1, 2, 4]`, and proves per reconnect that nothing respawns ten milliseconds short of each deadline.

## Assumptions

- The population of each fast pass stays in the same order of magnitude. The shared constants are derived from ~5,200 and ~4,800 tests; Tests/CLAUDE.md names them as the first thing to re-derive when the population moves materially.
- The fork-failure diagnostic assumes errno survives from SwiftTerm's failed `fork` to the statement after `startProcess` returns; if SwiftTerm adds work on that path the reading may be stale, and the message says where it was read.

## What remains, and one decision for a human

- **The latency itself is the lever, and bringing bounds to the derived constants does not lower it.** The repo's earlier measurement shows halving a pass's population halves the tail (p90 26.4 s to 14.6 s for +26 s wall). Pass 1 is a single target, so a further split means a `--filter` and its complementary `--skip` over suite names inside `TBDDaemonTests`, which the workflow's comments warn reintroduces the zero-match hazard unless written as a partition. Whether to pay that is a design choice; this PR does not make it.
- **Swift Testing's parallelization-width cap is live on the CI toolchain, and it is the decision this PR surfaces rather than makes.** Xcode 26.6's Testing binary carries it (confirmed with `strings` on the runner), and both routes reach it through SwiftPM, `scripts/swift-safe` and `scripts/test.sh`'s environment fence: `--experimental-maximum-parallelization-width N` on the command line, or `SWT_EXPERIMENTAL_MAXIMUM_PARALLELIZATION_WIDTH=N` in the step's environment. It caps how many test-case bodies run at once, which is exactly the population a bounded wait inside a body competes with. Measured with the event stream as the oracle (`testCaseStarted`/`testCaseEnded` pairs; an earlier read of the console log was wrong because the console never prints a test-case end line, so that counter could only climb):

  | arm | wall s | peak bodies in flight | p50 | p90 | p99 | > 90 s |
  |---|---|---|---|---|---|---|
  | pass 1 uncapped (green baseline) | 143.0 | 59 | 65.6 | 82.6 | 92.1 | 118 |
  | pass 1 uncapped (control run) | 180.8 | 59 | 77.4 | 102.8 | 114.6 | 1379 |
  | pass 1 width 12 | 145.2 | 10 | 48.7 | 82.9 | 88.5 | 44 |
  | pass 2 uncapped (control run) | 52.9 | 17 | 1.9 | 48.0 | 51.1 | 0 |
  | pass 2 width 4 | 75.5 | 3 | 3.4 | 61.2 | 68.7 | 0 |

  Width 12 on pass 1 cost nothing in wall time and cut the over-90 s tail from 118 to 44 while holding ten bodies in flight instead of fifty-nine; width 4 on pass 2 was too narrow and cost 43 %. The reported per-test durations still include time queued before the gate, so they understate what the cap does for a wait *inside* a body, which sees at most eleven competitors instead of ~5,000. Each arm is one sample; uncapped pass-1 wall time varied 143 to 265 s across runs. If you want it, it is one `env:` line on the fast-pass steps in `.github/workflows/test.yml`, CI-only (the local Xcode's Swift 6.2.4 lacks the feature and would ignore it); a value around 12 for pass 1 is the measured starting point and pass 2 has not been measured above 4.
- **This PR's own CI shows the ceiling of what bounds can do.** On one run of its final head (run 34003762464) pass 1 reported p50 108.9 s and p99 146.5 s per test, with 4581 of 5171 tests over 90 s, and four tests in files this branch had not touched went red at their 90 s waits. The three control-mode ones now run on the injected executor (above). The fourth, `WorktreeArchiveDeletionQueueTests.theFallbackRemovalArmsItsOwnRaisedTimeout`, is a different cause: its control assumes a 1 ms git deadline kills the removal before git completes, a subprocess race that meets the tier-3 criterion; the clean fix is moving that file to the live target at the cost of six more git-subprocess tests in the serial pass, left as a follow-up. On a runner that slow no fixed bound is sound, which is why the decision above is the one that matters.
- **Left alone on purpose:** `HolderReclaimTests` (`.exitedStatusUnknown` three times in five days; the registry code for that race changed in #807, #808 and #813 the same day, so re-check after a few days), and `UpdateLauncherTests`' one 20 s failure waiting on a spawned child's first log line, which is ambiguous between scheduling and the child.

## Evidence & verification

- Five days of CI logs (63 red `test` and harness jobs) mined by test name; the per-test distributions above come from the green reference job 101370987252 and reproduce with a one-line extraction over the log.
- The branch's first CI run (run 33999356107) ran 5171 tests in pass 1 with exactly one failure: the deliberately-hypothesised priority probe, whose diagnostic printed the four equal priorities. That failure is what the parity pin now asserts.
- The branch's second run (run 34000481421) passed both fast passes, 5171 and 4836 tests, with only the 71 and 11 deliberate known issues; the quiet pass was cancelled by the concurrency group and is re-run by this PR's own check. That green pass 1 reported p50 77.3 s, p99 108.6 s and 993 tests over 90 s, which is the saturated-pass constant sitting inside a healthy run's tail.
- The python fix is mutation-checked: against the pre-fix wrapper the harness reports 6 failures and 7 errors; with it, `Ran 61 tests`, OK. Reproduction rate before/after: 4/980 and 0/720 under 6-way parallel load in a 2-CPU container.
- The backoff test fails twice over against a flat backoff (requested delays `[1, 1, 1]`, and connection 4 opens during the third probe); the argument is analytic and stated in the test, since tier-3 tests cannot be run on this shared machine.
- `scripts/swift-safe build --build-tests` exit 0; `swiftlint --strict` 0 violations in 975 files. The CI verdict for the final branch state is in the checks below.
- Independent second opinion: a different model (Codex) was given the raw evidence and the sources and asked to reason from scratch. It agreed that priority parity refutes #815's rationale, that three to four hops at the measured latency explain "0 of 8 after 90 s", and with the classification of the fast failures; its one open point, why the width cap looked inert, was resolved by the event-stream measurement above (the console-log metric both of us had accepted was broken).
- Independent review: a `code-review` pass at high effort over the branch found no weakened assertion and no deleted coverage; its four findings (missing hang guard on the probe, a doc pointer to a claim that was not there, the refuted priority sentence, and the not-found window in the wrapper) are all addressed in later commits.

https://claude.ai/code/session_01MFnhK1b6kyiTLWBSWs3Lvf
https://claude.ai/code/session_01VmxVag4BeDdC65msZCawvK
[20260905-supreme-crayfish](https://cheapsteak.github.io/tbd/open/?worktree=71EC2F59-4AA3-449D-BB25-135365EF7FA3)

## Summary by CodeRabbit

- **Reliability**
  - Improved handling of shutdown signals to reduce orphaned processes, leaked sandboxes, and incomplete cleanup.
  - Preserved and reported multiple late-arriving termination signals across command execution paths.
  - Improved asynchronous teardown checks and diagnostics for failed process and terminal operations.

- **Testing**
  - Standardized asynchronous test timeouts and hang guards for more consistent results under scheduler contention.
  - Added coverage for reconnect backoff timing, task-priority behavior, executor-controlled input delivery, and signal-handling edge cases.

### CODE REVIEW

File: Tests/TBDAppTests/FDSidecarClientTests.swift [hunk 1]
```diff
@@ -465,6 +466,10 @@ struct FDSidecarClientTests {
         let promise = client.expectFD(worktreeID: UUID(), paneID: "%3", attachID: UUID())
         Darwin.close(daemonSide)   // daemon goes away
 
+        // Deliberately a short literal rather than the shared saturated-pass
+        // budget: the assertion here is that EOF fails the waiter, and the
+        // deadline only caps a run in which it did not. Raising it would make
+        // an already-failing test wait 90 s to say so.
```
[the remark this answers is not in the document]
Comment (cheapsteak) [reply Tests/TBDAppTests/FDSidecarClientTests.swift]: Agreed, and it was the more important of the three: `FDSidecarError.self` also matched `.timedOut`, so under load the test could pass without the behaviour. 894d681a requires exactly `.disconnected` (pattern-matched; the enum carries an errno payload so it is not Equatable), reports anything else on the primary failure line, moves the now-hang-guard deadline to the shared `TestDeadlines.saturatedPass`, and replaces the 200 ms sleep with a bounded poll.

File: Tests/TBDDaemonTests/ModelProfileSpawnTests.swift [hunk 2]
```diff
@@ -1499,9 +1499,16 @@ struct ModelProfileSpawnTests {
...
+    /// The deadline is the shared saturated-pass budget, not a literal: what it
+    /// waits for is produced by the coordinator's own detached pump task, which
+    /// runs on the cooperative pool behind the whole fast pass regardless of
+    /// how the test was started (`gateHoldingTask` in
+    /// `Tests/TestSupport/BoundedGateSupport.swift`). Five seconds is far below
+    /// that pass's healthy per-test latency.
     private func waitFor(
         _ condition: @Sendable () -> Bool,
-        timeout: Duration = .seconds(5)
+        timeout: Duration = TestDeadlines.saturatedPass
```
[the remark this answers is not in the document]
Comment (cheapsteak) [reply Tests/TBDDaemonTests/ModelProfileSpawnTests.swift:1511]: Done in c2c6f247: the loop returns the condition's current value as soon as `Task.isCancelled` after the swallowed sleep, so a cancelled task no longer spins through the deadline.

File: Tests/TBDDaemonLiveTests/PeerLinkSupervisorTests.swift [hunk 3]
```diff
@@ -119,73 +119,117 @@ struct PeerLinkSupervisorTests {
...
+                connection \(attempt + 1) opened before its backoff was up: the delay was \
+                computed as \(request.base)s and the clock is still short of it, so the \
+                supervisor must still be parked; spawns=\(stub.spawnCount())
+                """)
+
+            let opened = await advanceVirtualTime(
+                clock, until: "connection \(attempt + 1)",
+                observed: { "spawns=\(stub.spawnCount())" }
+            ) { stub.spawnCount() >= attempt + 1 }
+            if opened == nil { break }
```
[the remark this answers is not in the document]
Comment (cheapsteak) [reply Tests/TBDDaemonLiveTests/PeerLinkSupervisorTests.swift:216]: Good point about the upper bound; the probe only proved the supervisor waits at least the requested delay. Rather than add a second seam on the sleep, da0510bd asserts the observable: after `advanceVirtualTime` reports the connection open, `clock.now` must be within one driver step (0.25 s, now a single shared constant) of `request.at + delay`. With the lower-bound probe that pins the slept duration to [delay − 10 ms, delay + 0.25 s] per reconnect, for 1, 2 and 4 s.

[the remark this answers is not in the document]
Comment (cheapsteak) [reply Tests/TBDDaemonLiveTests/PeerLinkSupervisorTests.swift:216]: Reverting the upper bound: it failed on the first CI run at 5.48 s virtual against a 3.49 s ceiling. Once the backoff sleep fires, the reconnect's spawn is paced by real time, and the driver keeps advancing virtual time while other sleepers are armed — so "when the connection opened" measures the runner, which is exactly the term the rewrite removed. The load-independent proof is the lower-bound probe plus the exact [1, 2, 4] request sequence; a tight upper bound on spawn time cannot be made independent of the machine without a seam on the sleep itself, and that seam would only re-verify TestClock.

File: scripts/claude-stub.py [hunk 4]
```diff
@@ -512,6 +609,29 @@ def serve_until_signalled(
...
+    Every one of them, not the last: two stop signals in this window are two
+    facts about the run, and a summary that reports one is indistinguishable
+    from a run that only got one.
+    """
+    late = SIGNAL_TARGETS.late_signums
+    if not late:
+        return
+    noun = "signal" if len(late) == 1 else "signals"
+    rendered = ", ".join(str(signum) for signum in late)
+    report([f"{noun} {rendered} arrived while exiting; nothing left to stop"])
```
[the remark this answers is not in the document]
Comment (cheapsteak) [reply scripts/claude-stub.py:644]: Done in d5b45b3d, and the finding was understating it: `finishing` also leaked across in-process runs, so a second `main()` would have swallowed its own stop signals. The reset now happens at the top of `main` before the handlers are installed (`late_signums`, `unowned_signums`, `finishing`), with the other fields documented as self-resetting. The reporter is unchanged, as you suggested, so tests can still read the list after return. The single ownerless-signal slot from the review gate's minor is also a list now.
